### ISSUE TITLE
[MPOM-69] avoid Rat plugin warning

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-apache-parent#450]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** opened **[MPOM-69](https://issues.apache.org/jira/browse/MPOM-69?redirect=false)** and commented

see RAT-158

---

**Affects:** ASF-16

**Issue Links:**
- [RAT-158](https://issues.apache.org/jira/browse/RAT-158) SAXParser warnings / update parent pom to get rid of doxia dependency

- [ACCUMULO-2712](https://issues.apache.org/jira/browse/ACCUMULO-2712) Update Maven parent POM to latest version

**Comment by jira-importer** [comment]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

fixed in [r1659746](http://svn.apache.org/r1659746)

**Comment by jira-importer** [comment]
**[Sebb](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sebb@apache.org)** commented

The comment in the Apache Pom refers to RAT-158.

However RAT is not the cause of the problem. The cause is Doxia - RAT merely exposes the issue.
So the comment should refer to

http://jira.codehaus.org/browse/DOXIA-526

instead.

**Comment by jira-importer** [comment]
**[Sebb](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sebb@apache.org)** commented

Furthermore, the patch to the Apache pom should be flagged as a temporary work-round for the Doxia issue.

Once a fixed Doxia is released, the Apache pom needs to be updated to use the new version of Doxia.
The exclusion then becomes unnecessary.

**Comment by jira-importer** [comment]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

in fact, it's both Doxia and Rat fault: Doxia prepared a misleading situation where Rat failed
and in the end, Rat could/should have excluded the dependency: definitely RAT-158 makes the issue more clear (if not the final root cause)

the comment is not about root cause: it's about the end-user issue, which is at Rat level

**Comment by jira-importer** [comment]
**[Sebb](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sebb@apache.org)** commented

The warning is nothing to do with RAT per se.
It just so happens that using RAT exposes the problem.

It could well be that other plugins show the same problem.
RAT is very commonly used, so it's not suprising that the problem was noticed when using RAT.
At the very least the Apache pom comment should note the underlying cause.

**Comment by jira-importer** [comment]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

upgrading to Doxia 1.5+ will break Rat Maven plugin, because of https://jira.codehaus.org/browse/DOXIASITETOOLS-85

I don't really know what you expect from "flagging as a temporary work-round for the Doxia issue": you seem to look for a culprit

**Comment by jira-importer** [comment]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

it has to do with Rat maven plugin, not the underlying Rat core, true

**Comment by jira-importer** [comment]
**[Sebb](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sebb@apache.org)** commented

The problem obviously only occurs when the Maven RAT plugin is used since it relates to the jars which are automatically loaded by Maven.
However surely the same error could be reported by other similar Maven plugins?

The reasons I think the Apache pom comment needs to document the underlying Doxia problem are as follows:
* other plugins may potentially be affected, not just RAT
* if/when the Doxia issue is fixed, the work-round can be dropped

I think the comment should say something like the following:

\\

**Comment by jira-importer** [comment]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

> it relates to the jars which are automatically loaded by Maven
> it relates to the jars selected by Maven RAT plugin

the work-around is really for RAT-158

I don't even know if anybody will read the comment: what everybody is expecting is that Rat maven plugin works nicely by default
With the work-around in ASF parent, it will work

to make you happy, I can add link to DOXIA-526 too: seriously childish

**Comment by jira-importer** [comment]
**[Sebb](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sebb@apache.org)** commented

> it relates to the jars selected by Maven RAT plugin

If this problem only affects RAT, why update the Apache pom at all?

**Comment by jira-importer** [comment]
**[Herve Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

because a lot of Apache project use ASF parent, then implementing the workaround in ASF parent's pluginManagement automagically makes Apache projects avoid the problem before Maven RAT plugin is fixed: don't really need to know if the problem if fixed in Maven RAT plugin or through Maven RAT plugin configuration in ASF parent

**Comment by jira-importer** [comment]
**[Sebb](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sebb@apache.org)** commented

You are assuming that the Apache pom will be released before RAT, and that projects are willing and able to update the Apache pom.

Also, I have just realised that the fix is in the pluginManagement section.
So the fix to the Apache pom will only fix command-line invocation, it won't affect RAT report generation as part of the site build.
I've just tried, and mvn site still reports the warnings.
